src/storage/FileResourceStore.ts

```diff
@@ -0,0 +1,467 @@
+    }
+    if (stats.isFile()) {
+      return await this.getFileRepresentation(path, stats);
+    }
+    if (stats.isDirectory()) {
+      return await this.getDirectoryRepresentation(ensureTrailingSlash(path), stats);
+    }
+    throw new NotFoundHttpError();
+  }
+
+  /**
+   * @throws Not supported.
+   */
+  public async modifyResource(): Promise<void> {
+    throw new Error('Not supported.');
+  }
+
+  /**
+   * Replaces the stored Representation with the new one for the given identifier.
+   * @param identifier - Identifier to replace.
+   * @param representation - New Representation.
+   */
+  public async setRepresentation(identifier: ResourceIdentifier, representation: Representation): Promise<void> {
+    if (representation.dataType !== DATA_TYPE_BINARY) {
+      throw new UnsupportedMediaTypeHttpError('FileResourceStore only supports binary representations.');
+    }
+
+    // Break up the request URI in the different parts `path` and `slug` as we know their semantics from addResource
+    // to call the InteractionController in the same way.
+    const [ , path, slug ] = /^(.*\/)([^/]+\/?)?$/u.exec(this.parseIdentifier(identifier)) ?? [];
+    if ((path === undefined || normalizePath(path) === '/') && slug === undefined) {
+      throw new ConflictHttpError('Container with that identifier already exists (root).');
+    }
+    const { raw } = representation.metadata;
+    const linkTypes = representation.metadata.linkRel?.type;
+    let metadata: Readable | undefined;
+    if (raw.length > 0) {
+      metadata = streamifyArray(raw);
+    }
+
+    const isContainer = this.interactionController.isContainer(slug, linkTypes);
+    const newIdentifier = this.interactionController.generateIdentifier(isContainer, slug);
+
+    return isContainer ?
+      await this.setDirectoryRepresentation(path, newIdentifier, metadata) :
+      await this.setFileRepresentation(path, newIdentifier, representation.data, metadata);
+  }
+
+  /**
+   * Strips the baseRequestURI from the identifier and checks if the stripped base URI matches the store's one.
+   * @param identifier - Incoming identifier.
+   *
+   * @throws {@link NotFoundHttpError}
+   * If the identifier does not match the baseRequestURI path of the store.
+   */
+  private parseIdentifier(identifier: ResourceIdentifier): string {
+    if (!identifier.path.startsWith(this.baseRequestURI)) {
+      throw new NotFoundHttpError();
+    }
+    return identifier.path.slice(this.baseRequestURI.length);
+  }
+
+  /**
+   * Strips the rootFilepath path from the filepath and adds the baseRequestURI in front of it.
+   * @param path - The filepath.
+   *
+   * @throws {@Link Error}
+   * If the filepath does not match the rootFilepath path of the store.
+   */
+  private mapFilepathToUrl(path: string): string {
+    if (!path.startsWith(this.rootFilepath)) {
+      throw new Error('Cannot map filepath to URL.');
+    }
+    return this.baseRequestURI + path.slice(this.rootFilepath.length);
```

Comment:
`substr` instead of `slice`, and perhaps use string interpolation rather than `+`?

Reply:
So this is going to be another linter comment, but the linter recommends using `slice` instead of `substr` :D. I actually also thought it enfored string interpolation so surprised it doesn't complain about that (maybe because there are no string literals on this line).

Reply:
My bad.